### ISSUE TITLE
MacOSX - Opening Caret Window Size

### ISSUE THREAD

**Issue by srayhunter** [careteditor/caret#363]
Every time I open caret on mac I have to resize the window. It always opens in a small window.

Is there a way to have it open to the previous size?

**Comment by astoilkov** [comment]
Hi @srayhunter,

Caret should save the window size and restore it when you reopen the app. Maybe there is a problem somewhere. Could you describe the steps for reproducing the problem? What version of macOS are you on?

Thanks in advance.

**Comment by srayhunter** [comment]
Reproduce: I open caret, usually from the commandline, and then will have to resize it. Then after I close the app and open it again, I will have to resize the window again.

Mac: 10.12.4
Caret: 2.0.11

**Comment by astoilkov** [comment]
@srayhunter - I have tried to replicate the problem on the same version of macOS but without success. Does your other settings like font, font size, line width or line height persist?

One additional question: Do you close the window using the (x) button on the top left of the window before quitting the app?

**Comment by srayhunter** [comment]
@astoilkov so when I close the app I only really use `command-w`. It is rare that I ever close it down by clicking the `x` button on the top left of the window.

The other settings seems to be persisted after closing the app.

After trying out the `x` button the window will open with the same size until I use close it down with `command-w`. I think by closing the app with `command-w` it does not keep the window size.

**Comment by srayhunter** [comment]
@astoilkov when I save the preferences those are saving periodically, there are times where I will change the font size or line height and there is a message about saving the values. This comes up all the time, but if I save it again the values will be saved and the error message will go away. This is happening all the time when I change the preferences; yet, I do not change them that much.

Here is a screenshot of the error, but the values have been saved. In this screenshot, I changed the font-size to 14 from 15. The error message came up, but the value was changed and saved.

![caret-save-prefs-error](https://cloud.githubusercontent.com/assets/68721/26060859/866e1ff4-3943-11e7-9fdf-a471cfb68312.png)
